src/sidebar/store/modules/annotations.ts

```diff
@@ -3,46 +3,85 @@
  * sidebar.
  */
 
+import type { Dispatch } from 'redux';
 import { createSelector } from 'reselect';
 
 import { hasOwn } from '../../../shared/has-own';
+import type { Annotation, SavedAnnotation } from '../../../types/api';
+import type { HighlightCluster } from '../../../types/shared';
+
 import * as metadata from '../../helpers/annotation-metadata';
 import { isHighlight, isSaved } from '../../helpers/annotation-metadata';
 import { countIf, toTrueMap, trueKeys } from '../../util/collections';
 import { createStoreModule, makeAction } from '../create-store';
 
 import { routeModule } from './route';
+import type { State as RouteState } from './route';
 import { sessionModule } from './session';
+import type { State as SessionState } from './session';
 
-/**
- * @typedef {'anchored'|'orphan'|'timeout'} AnchorStatus
- * @typedef {import('../../../types/api').Annotation} Annotation
- * @typedef {import('../../../types/shared').HighlightCluster} HighlightCluster
- * @typedef {import('../../../types/shared').ClientAnnotationData} ClientAnnotationData
- * @typedef {import('../../../types/api').SavedAnnotation} SavedAnnotation
- */
+type AnchorStatus = 'anchored' | 'orphan' | 'timeout';
 
-/**
- * @typedef AnnotationStub
- * @prop {string} [id] - service-provided identifier if annotation has been
- *       persisted to the service
- * @prop {string} [$tag] - local-generated identifier
- */
+type AnchorStatusUpdates = {
+  [$tag: string]: AnchorStatus;
+};
+
+type AnnotationStub = {
+  /**
+   * service-provided identifier if annotation has been
+   * persisted to the service
+   */
+  id?: string;
+
+  /** local-generated identifier */
+  $tag?: string;
+};
+
+const initialState = {
+  annotations: [],
+  highlighted: {},
+  hovered: {},
+  nextTag: 1,
+} as {
+  /** Set of currently-loaded annotations */
+  annotations: Annotation[];
+
+  /**
+   * The $tags of annotations that should appear as "highlighted", e.g. the
+   * target of a single-annotation view. NB: This feature is currently not
+   * supported in the application UI.
```

Comment:
If `highlighted` is not used in the UI, what is it used for?

Reply:
This is the reason I did this work in the first place: it's not used for _anything_, as such.

See https://github.com/hypothesis/client/blob/3fff4b5e8891e253d61070d56a7d1fdd452fbdd9/src/sidebar/components/AnnotationView.tsx#L64-L68

The _intention_ is that there is a visual distinction of the actual annotation corresponding to the single-annotation view's URL if it is not the top-level annotation. But this hasn't "worked" in some time in the sense that no visual styles ever get applied.

My intent is to:

* Rename this (I don't think we should call it _highlight_, do you?)
* Make it work again by finding a good design to apply in this case

As part of a broader think about different UI states an "annotation" can be in.